gitlab-runner-docker/Dockerfile

```diff
@@ -0,0 +1,38 @@
+#
+# Dockerfile for a system container that includes the GitLab runner
+# plus it's own docker daemon.
+#
+# This allows the GitLab runner to run with the Docker executor in total
+# isolation from the underlying host, thus improving security (particularly for
+# CI jobs that interact with Docker by issuing commands such as "docker build"
+# or "docker run".
+#
+# Start the GitLab runner with:
+#
+# $ docker run --runtime=sysbox-runc -d --name gitlab-runner --restart always -v /srv/gitlab-runner/config:/etc/gitlab-runner nestybox/gitlab-runner-docker
+#
+# Then register it with the GitLab server:
+#
+# $ docker run --rm -it -v /srv/gitlab-runner/config:/etc/gitlab-runner gitlab/gitlab-runner register
+#
+
+FROM gitlab/gitlab-runner:latest
```

Comment:
Couple of minor suggestions that may reduce this image size:

1) Pair every "apt-get update" with a "rm -rf /var/lib/apt/lists/*" instruction.

2) Reduce number of layers in cases where it makes sense such as "apt-get bla && curl bla && apt-get repo".

Reply:
I added the apt cleanup, thanks!.

I left the apt-* as they were since they can't be merged into a single command due to dependencies.